src/pam_hbac_obj.c

```diff
@@ -61,6 +64,10 @@ getgroupname(gid_t gid)
         return NULL;
     }
 
+#ifndef getgrgid_r               // to avoid not compiling on hpux
+  extern int getgrgid_r(gid_t, struct group *, char *, size_t, struct group**);
+#endif
+
```

Comment:
HP-UX has `getgrgid_r`, but I get an implicit declaration error when compiling:

> libtool: compile:  gcc -DHAVE_CONFIG_H -I. -Wall -I./src -I. -DPAM_HBAC_CONF_DIR=\"/etc\" -Wall -Wshadow -Wstrict-prototypes -Wpointer-arith -Wcast-qual -Wcast-align -Wwrite-strings -Werror-implicit-function-declaration -I/usr/local/include/glib-2.0 -I/usr/local/lib/hpux32/glib-2.0/include -I/usr/local/include -g -O2 -MT src/pam_hbac_obj.lo -MD -MP -MF src/.deps/pam_hbac_obj.Tpo -c src/pam_hbac_obj.c  -fPIC -DPIC -o src/.libs/pam_hbac_obj.o
> src/pam_hbac_obj.c: In function 'getgroupname':
> src/pam_hbac_obj.c:65: error: implicit declaration of function 'getgrgid_r'
> src/pam_hbac_obj.c:141:2: error: #error No known get-groups-for-user implementation found
> src/pam_hbac_obj.c: In function 'get_user_groups':
> src/pam_hbac_obj.c:127: warning: unused variable 'ngroups'
> src/pam_hbac_obj.c: In function 'get_user_int':
> src/pam_hbac_obj.c:158: error: implicit declaration of function 'getpwnam_r'
> *** Error exit code 1
> 
> Stop.
> 

Here's the config.log from `LDFLAGS="-L/usr/local/lib/hpux32/" ./configure --with-pammoddir=/usr/lib/security/hpux32 --mandir=/usr/share/man --sysconfdir=/etc --mandir=/usr/local/share/man
`

[config.log.txt](https://github.com/jhrozek/pam_hbac/files/500280/config.log.txt)